opslevel/resource_opslevel_tag.go

```diff
@@ -0,0 +1,162 @@
+package opslevel
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/opslevel/opslevel-go/v2023"
+)
+
+func resourceTag() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a tag. Only uses API's 'tag create', not 'tag assign'.",
+		Create:      wrap(resourceTagCreate),
+		Read:        wrap(resourceTagRead),
+		Update:      wrap(resourceTagUpdate),
+		Delete:      wrap(resourceTagDelete),
+		Schema: map[string]*schema.Schema{
+			"last_updated": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"resource_id": {
+				Type:        schema.TypeString,
+				Description: "The id or human-friendly, unique identifier of the resource this tag belongs to.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"resource_type": {
+				Type:         schema.TypeString,
+				Description:  "The resource type that the tag applies to.",
+				ForceNew:     true,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(opslevel.AllTaggableResource, false),
+			},
+			"key": {
+				Type:        schema.TypeString,
+				Description: "The key of the tag.",
+				ForceNew:    false,
+				Optional:    true,
+			},
+			"value": {
+				Type:        schema.TypeString,
+				Description: "The value of the tag.",
+				ForceNew:    false,
+				Optional:    true,
```

Comment:
.... should this field be required?  I guess what happens if the customer doesn't set this field.  Can we create a tag with an empty value?

Reply:
This should be required. I just tried to create a tag with an empty value and got an API Error saying `'value' Value is too short (minimum is 1 character)`